examples/federated-diffusion/README.md

```diff
@@ -0,0 +1,888 @@
+
+        return loss_aggregated, metrics_aggregate
+```
+
+Note that the Federated Learning for diffusion model can be implemented with a
+standard `FedAvg` strategy, this custom strategy is only here to bring extra
+features (like threshold filtering).
+
+### Starting the simulation
+
+In order to start the simulation, we first instantiate our strategy:
+
+```python
+strategy = SaveModelAndMetricsStrategy(
+    fraction_fit=1.0,  # Sample 100% of available clients for training
+    fraction_evaluate=0.5,  # Sample 50% of available clients for evaluation
+    min_fit_clients=PARAMS.num_clients,  # Never sample less than 10 clients for training
+    min_evaluate_clients=PARAMS.num_clients,  # Never sample less than 5 clients for evaluation
+    min_available_clients=PARAMS.num_clients,  # Wait until all 10 clients are available
+    on_fit_config_fn=trainconfig,
+    on_evaluate_config_fn=trainconfig,
+    client_manager=client_manager,
+)
+```
+
+Note that we are using a custom `client_manager` and a `trainconfig` function defined here:
+
+```python
+def trainconfig(server_round):
+    """Return training configuration dict for each round."""
+    config = {"server_round": server_round}  # The current round of federated learning
+    return config
+
+
+class ClientManager(fl.server.SimpleClientManager):
+    def sample(
+        self,
+        num_clients,
+        server_round,
+        min_num_clients,
+        criterion,
+    ):
+        """Sample a number of Flower ClientProxy instances."""
+        # Block until at least num_clients are connected.
+        if min_num_clients is None:
+            min_num_clients = num_clients
+        self.wait_for(min_num_clients)
+        # Sample clients which meet the criterion
+        available_cids = list(self.clients)
+
+        # First round and odd rounds
+        if server_round >= 1 and num_clients <= 20:
+            available_cids_sorted = sorted(available_cids, key=int)
+            section = []
+            for j in range(4):
+                index = ((server_round - 1) * 4 + j) % len(available_cids_sorted)
+                section.append(available_cids_sorted[index])
+            available_cids = section.copy()
+            print("Available cids: ", available_cids)
+            return [self.clients[cid] for cid in available_cids]
+
+        if num_clients > 20:
+            print(
+                "Support for distributed training of UNet in specified hardware for larger clients is unavailable"
+            )
+```
+
+Then, once all of this is defined, we can finally call our `start_simulation` function:
+
+```python
+# Start simulation
+fl.simulation.start_simulation(
+    client_fn=client_fn,
+    num_clients=PARAMS.num_clients,
+    # client_resources={"num_cpus": 10, "num_gpus":1},
+    config=fl.server.ServerConfig(num_rounds=PARAMS.num_rounds),
+    strategy=strategy,
+)
+```
+
+We commented out the `client_resources` parameters as it will depend on each person's setup.
```

Comment:
```suggestion
We commented out the `client_resources` parameters as it will depend on each person's setup. Read more about how to set these up in the [Documentation](https://flower.dev/docs/framework/how-to-run-simulations.html#virtualclientengine-resources).
```